src/Workspaces/Remote/Core/RemoteHostAssetSerialization.cs

```diff
@@ -125,7 +125,7 @@ public static async ValueTask<ImmutableArray<(Checksum, object)>> ReadDataAsync(
                 using var stream = localPipe.Reader.AsStream(leaveOpen: false);
                 return ReadData(stream, scopeId, checksums, serializerService, cancellationToken);
             }
-            catch (EndOfStreamException)
+            catch (EndOfStreamException e) when (IsExpectedEndOfStreamException(e, exception, cancellationToken))
```

Comment:
is there any need to do anything similar to this in the copyTask above?  Overall, i have a tiny concern about the two different patterns we see here for writing vs reading.  In my ideal world those would be near-identical/reflective.  If that's not the case, that's fine.  I just wanted to get some confirmation given the issues we've had here.  Thanks!

Reply:
No, exceptions in the copy operation are written to the pipe, and subsequently observed when they impact a read.